instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/routeline/AlternativeRouteSelectionTest.kt

```diff
@@ -42,124 +34,111 @@ class AlternativeRouteSelectionTest : BaseTest<BasicNavigationViewActivity>(
     @get:Rule
     val mockLocationReplayerRule = MockLocationReplayerRule(mockLocationUpdatesRule)
 
-    private lateinit var initIdlingResource: MapStyleInitIdlingResource
-    private lateinit var routeProgressIdlingResource: RouteProgressStateIdlingResource
-
-    protected lateinit var directionsResponse: DirectionsResponse
-
-    protected lateinit var mapboxNavigation: MapboxNavigation
-
-    protected lateinit var routeLineApi: MapboxRouteLineApi
-    protected lateinit var routeLineView: MapboxRouteLineView
-
-    private val myResourceIdler =
-        CountingIdlingResource("AlternativeRouteSelectionTestResource")
+    private lateinit var mapboxNavigation: MapboxNavigation
+    private lateinit var routeLineApi: MapboxRouteLineApi
+    private lateinit var routeLineView: MapboxRouteLineView
 
     @Before
     fun setUp() {
-        initIdlingResource = MapStyleInitIdlingResource(activity.binding.mapView)
-        initIdlingResource.register()
-        IdlingRegistry.getInstance().register(myResourceIdler)
-        Espresso.onIdle()
-
-        directionsResponse = getRoute(activity)
-        val coordinates = directionsResponse.waypoints()?.mapNotNull { it.location() } ?: listOf()
-        val route = directionsResponse.routes()[0]
-
         mapboxNavigation = MapboxNavigation(
             NavigationOptions.Builder(activity)
                 .accessToken(getMapboxAccessTokenFromResources(activity))
                 .build()
         )
-
-        routeProgressIdlingResource = RouteProgressStateIdlingResource(
-            mapboxNavigation,
-            RouteProgressState.TRACKING
-        )
-
-        runOnMainSync {
-            mockLocationUpdatesRule.pushLocationUpdate {
-                latitude = coordinates.first().latitude()
-                longitude = coordinates.first().longitude()
-            }
-            mockLocationReplayerRule.playRoute(route)
-
-            mapboxNavigation.setRoutes(directionsResponse.routes())
-        }
     }
 
     @After
     fun tearDown() {
-        initIdlingResource.unregister()
+        mapboxNavigation.onDestroy()
```

Comment:
While working on https://github.com/mapbox/mapbox-navigation-android/issues/4429 along with https://github.com/mapbox/mapbox-navigation-android/pull/4419 and https://github.com/mapbox/mapbox-navigation-android/pull/4507 noticed that the `@Rule` was not doing anything because `MapboxNavigation` was already destroyed by the time the `@Rule` was executed e.g. https://github.com/mapbox/mapbox-navigation-android/blob/25fe3744b9ab81c9d80622af108e7b65eb3ccf79/instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/routeline/AlternativeRouteSelectionTest.kt#L137 Do you know if the `@Rule` is executed before `@After`? If not, this may be a problem if the `@Rule` is added to the test. Not a problem now because this test is not adding it but was wondering if we could establish a "pattern" for other tests to mimic and avoid any issues when _copy & paste_ 😅

Reply:
You're right, thanks for flagging. `MapboxNavigation` will be destroyed before saving the history file which is not what we would want. That makes me feel like the `MapboxHistoryTestRule` is not how we should save history files. 

``` kotlin
    val initializer = initializer()

    @get:Rule
    val logTestRule = LogTestRule1()

    @get:Rule
    val logTestRule = LogTestRule2()

    @Before
    fun setUp() {
        Log.i("UnitTest", "setUp")
    }

    private fun initializer(): Long {
        Log.i("UnitTest", "initializer")
        return SystemClock.elapsedRealtimeNanos()
    }

    @After
    fun tearDown() {
        Log.i("UnitTest", "tearDown")
        mapboxNavigation.onDestroy()
    }

    @Test
    fun run_the_test() {
        Log.i("UnitTest", "run_the_test")
    }
```

```
UnitTest initializer
  LogTestRule1 starting
    LogTestRule2 starting
      UnitTest setUp
        UnitTest run_the_test
      UnitTest tearDown
    LogTestRule2 finished
  LogTestRule1 finished
```

Reply:
I mean, if `MapboxNavigation` is recreated on every test, it shouldn't be a problem to not call `onDestroy` ¯\\\_(ツ)\_/¯